lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -615,38 +476,50 @@ static void reconnectMergeOps(Region &r,
   removeBlockOperands(r);
 }
 
+static bool isAllocOp(Operation *op) {
+  return isa<memref::AllocOp, memref::AllocaOp>(op);
+}
+
+LogicalResult
+HandshakeLowering::maximizeSSANoMem(ConversionPatternRewriter &rewriter) {
```

Comment:
It would be nice if we could somehow pass in a lambda or a "strategy class" to the other function to make this customizable.

Reply:
This would make the utilities more general indeed. We could change the utilities' API (and their implementation to apply SSA maximization only when filters return `true`) to something like the following for example.
```cpp
maximizeSSA(Value value, PatternRewriter &rewriter));
maximizeSSA(Operation* op, PatternRewriter &rewriter, bool(Operation*) filter));
maximizeSSA(Block* block, PatternRewriter &rewriter, bool(BlockArgument) argFilter, bool(Operation*) opFilter)));
maximizeSSA(Region& region, PatternRewriter &rewriter, bool(Block*) blockFilter, bool(BlockArgument) argFilter, bool(Operation*) opFilter)));
```
Perhaps with default filters that always return `true`?

Reply:
That looks like the right approach. I would probably implement this as a class that defines a set of member functions to pack this logically together and to reduce the amount of parameters.

Reply:
Sounds good, will make the change.